test/adapters/level_zero/multi_device_event_cache_tests.cpp

```diff
@@ -0,0 +1,107 @@
+// Copyright (C) 2024 Intel Corporation
+// Part of the Unified-Runtime Project, under the Apache License v2.0 with LLVM Exceptions.
+// See LICENSE.TXT
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+
+#include "ur_print.hpp"
+#include "uur/fixtures.h"
+#include "uur/raii.h"
+
+#include <map>
+#include <string>
+
```

Comment:
This is just testing whether the implementation calls the L0 calls we expect it to. Those sort of tests tend to be fragile. Would it be possible to instead create a test that verifies the expected outcome (i.e., the commands are ordered correctly on multi-device systems)?